lib/src/replay/screenshot/screenshot_capturer.dart

```diff
@@ -47,37 +51,33 @@ class ScreenshotCapturer {
     return min(width / srcWidth, height / srcHeight);
   }
 
-  void _updateStatusView(bool shouldSendMetaEvent, RenderObject renderObject,
-      ViewTreeSnapshotStatus statusView) {
-    if (shouldSendMetaEvent) {
-      statusView.sentMetaEvent = true;
-    }
-    _views[renderObject] = statusView;
-  }
-
   Future<Uint8List?> _getImageBytes(ui.Image img) async {
-    final ByteData? byteData =
-        await img.toByteData(format: ui.ImageByteFormat.png);
-    if (byteData == null || byteData.lengthInBytes == 0) {
-      printIfDebug('Error: Failed to convert image to byte data.');
+    try {
+      final ByteData? byteData =
+          await img.toByteData(format: ui.ImageByteFormat.png);
+      if (byteData == null) {
+        printIfDebug('Error: Failed to convert image to byte data.');
+        return null;
+      }
+      return byteData.buffer.asUint8List();
+    } catch (e) {
+      printIfDebug('Error converting image to byte data: $e');
       return null;
     }
-    return byteData.buffer.asUint8List();
   }
 
-  Future<ImageInfo?> captureScreenshot() async {
+  Future<ImageInfo?> captureScreenshot() {
```

Comment:
the function has to be async if it returns a Future, right?

Reply:
Not necessarily, a function does not need to be explicitly marked as async to return a Future.

In our case, we need to work synchronously with the function. Then we execute the main process asynchronously as a "task". Managing this with the Completer. When this is ready, we retrieve the Completer and return the Future.

This way, the frame image and the masks always move in the main thread.

Reply:
true that, nice